virttest/qemu_vm.py

```diff
@@ -1598,6 +1599,29 @@ def sort_key(dev):
         add_memorys(devices, params)
         mem = int(params.get("mem", 0))
 
+        if params.get("auto_cpu_model") == "yes":
+            policy_map = {"libvirt_host_model": cpu.get_cpu_info_from_virsh,
+                          "virttest": cpu.get_qemu_best_cpu_info}
+            auto_cpu_policy = params.get("auto_cpu_policy", "virttest").split()
+            for policy in auto_cpu_policy:
+                try:
+                    cpu_info = policy_map[policy](params)
+                    if cpu_info:
+                        break
+                except Exception as err:
+                    logging.error("Failed to get cpu info with policy %s: %s"
+                                  % (policy, err))
+                    continue
+            else:
+                raise exceptions.TestCancel("Failed to get cpu info with "
+                                            "policy %s" % auto_cpu_policy)
+
+            params["cpu_model"] = cpu_info["model"]
```

Comment:
Looks here would be a use-before-define error if the procedure being performed at this stage.

https://github.com/avocado-framework/avocado-vt/blob/9c2bc1139c163dd9e78cf4ac51adce47e4da2b07/virttest/qemu_vm.py#L1547
https://github.com/avocado-framework/avocado-vt/blob/9c2bc1139c163dd9e78cf4ac51adce47e4da2b07/virttest/qemu_devices/qcontainer.py#L1149

Reply:
Seems so.  But I didn't run into this error with either q35 or i440fx machine type. I wonder what's the QCPUBUS for, seems it's only used in these machine functions.
BTW, with configuration ''auto_cpu_model = no cpu_model = ", it goes to the same situation.

Reply:
How about move L1602~L1654 before L1548? That should solve the issue.